pkg/storage/cmdq/interval_btree.go

```diff
@@ -589,10 +589,12 @@ func (t *btree) Set(c *cmd) {
 	}
 }
 
-// MakeIter returns a new iterator object. Note that it is safe for an
-// iterator to be copied by value.
+// MakeIter returns a new iterator object. Note that it is safe for an iterator
+// to be copied by value. However, it is not safe to make modification to the
```

Comment:
Is that really true? If you create an iterator, use it until it uses `s` and then copy it by value, won't the copy and the original share its `s` which gets mutated by each iterator?